sdks/go/pkg/beam/core/runtime/exec/util.go

```diff
@@ -33,13 +34,29 @@ func (g *GenID) New() UnitID {
 	return UnitID(g.last)
 }
 
+type doFnError struct {
+	doFn string
+	err  error
+	uid  UnitID
+	pid  string
+}
+
+func (e *doFnError) Error() string {
+	return fmt.Sprintf("DoFn[<%d>;<%s>]<%s> returned error:<%s>", e.uid, e.pid, e.doFn, e.err)
+}
+
 // callNoPanic calls the given function and catches any panic.
 func callNoPanic(ctx context.Context, fn func(context.Context) error) (err error) {
 	defer func() {
 		if r := recover(); r != nil {
-			// Top level error is the panic itself, but also include the stack trace as the original error.
-			// Higher levels can then add appropriate context without getting pushed down by the stack trace.
-			err = errors.SetTopLevelMsgf(errors.Errorf("panic: %v %s", r, debug.Stack()), "panic: %v", r)
+			// check if error is of type DoFn then handle it appropriately
```

Comment:
Please change this comment to
```suggestion
			// Check if the panic value is from a failed DoFn, and return it without a panic trace.
```

A comment saying 'handle it appropriately' doesn't communicate anything to the reader. What does that mean?

Reply:
Right, should have explained it more clearly